cs_api_core/cs_api_core.ml

```diff
@@ -191,7 +191,7 @@ let parse_s3_signature_request ~body =
 
 let parse_s3_response ~body =
   try
-    let key_extractor = Str.regexp "<Key>\\([^<>]*\\)</Key>" in
+    let key_extractor = Str.regexp "<.*Key.*>\\([^<>]*\\)</.*Key.*>" in
```

Comment:
**Suggestion (non-blocking):** Use a stricter regexp

From what I understand and from the example you use in your test, the xml element is always named `Key`, it's just that sometimes  there's also a `xmlns` attribute attached to it.

How about something like that:
```suggestion
    let key_extractor = Str.regexp "<Key.*>\\([^<>]*\\)</Key>" in
```

or even stricter:
```suggestion
    let key_extractor = Str.regexp "<Key\\( xmlns=.+\\)?>\\([^<>]*\\)</Key>" in
```
(in which case we would need to return the second matched group).

What do you think?

Reply:
The issue is that we are doing «cowboy» XML parsing. As such, `<Key xmlns="foo">`, `<Key another="bar" xmlns="foo">` are both valid and should both work. I have opted for the middle ground solution in your first suggestion. An XML validator told me that `<xmls="foo" Key>` is not valid, so its probably fine not to support it.